packages/styles/components/_c.button.scss

```diff
@@ -35,6 +35,9 @@
   }
 
   &__icon {
+    width: #{map-get($button-icon-default-size, 'width')};
```

Comment:
why not just do : 

```scss
$button-sizes: (
  's': (
    'font-size': '04',
    'line-height': 'xs',
    'padding-v': $mu025,
    'padding-h': $mu100,
    'min-height': $mu200,
    'icon-size': $mu150,
    'only-icon-size': $mu150,
  ),
  'm': (
    'font-size': '05',
    'line-height': 'xs',
    'padding-v': $mu025,
    'padding-h': $mu200,
    'min-height': $mu300,
    'icon-size': $mu150,
     'only-icon-size': $mu200,
  ),
  'l': (
    'font-size': '06',
    'line-height': 'xs',
    'padding-v': $mu025,
    'padding-h': $mu200,
    'min-height': $mu350,
    'icon-size': $mu200,
     'only-icon-size': $mu250,
  ),
);
```

note the **icon-size** and **'only-icon-size'** keys


and remove the icon map.
If icons sizes are tied to button size, why letting the user the choice for a size modifier on the icon ?

```scss
.mc-button {
    /*...*/

    &--l &__icon {
       width : $icon-size
       height : $icon-size
    }

    &--square&--l &__icon {
       width : $only-icon-size
       height : $only-icon-size
    }
}
```

Reply:
fixed